pallets/crowdloans/src/lib.rs

```diff
@@ -902,6 +957,8 @@ pub mod pallet {
                         ChildStorageKind::Flying,
                     )?;
                     Vaults::<T>::insert(crowdloan, vault.id, vault);
+
+                    Self::deposit_event(Event::<T>::VaultContributeFailed(crowdloan, who, amount));
```

Comment:
we just need to record VaultContributeSucceed event I think because the failed one wont be useful. It's already left in XcmInflight

Reply:
okay